File: AquariumApi.Models/Device/DeviceAction.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace AquariumApi.Models
{
    public class DeviceAction
    { 
        public int? Id { get; set; } = null;
        public int TargetSensorId { get; set; }
        public GpioPinValue TargetSensorValue { get; set; }

        [ForeignKey("TargetSensorId")]
        public virtual DeviceSensor TargetSensor { get; set; }
    }
}

File: AquariumApi.DeviceApi/ClientApp/src/app/modules/SharedDeviceModule/models/DeviceAction.ts
import { DeviceScheduleTaskTypes } from "../../../models/types/DeviceScheduleTaskTypes";
import { DeviceSensor } from "./DeviceSensor";
import { GpioPinValue } from "./GpioPinValue";

export class DeviceAction {
  id: number; //= Math.floor(Math.random() * 100);
  targetSensorId: number;
  targetSensorValue: GpioPinValue;
  targetSensor: DeviceSensor;
}
